src/components/Tree/Tree.tsx

```diff
@@ -143,6 +141,7 @@ const reducer = produce((draft: TreeState, action: TreeStateAction) => {
                     return;
                 }
 
+                const parentNodeIndex = findIndexById(draft.rootNodes, parentId);
```

Comment:
Why to change the way it was done before? With this change, the `findIndexById` will be called twice instead of once. Was there any bug related to that?

Reply:
no reason, it was left from a experiment... i will revert it 
nice catch!